pkg/compactor/blocks_cleaner.go

```diff
@@ -321,15 +326,40 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userID string, firstRun b
 		}
 	}
 
+	// Reading bucket index sync stats
+	idxs, err := bucketindex.ReadSyncStatus(ctx, c.bucketClient, userID, userLogger)
+
+	if err != nil {
+		level.Warn(userLogger).Log("msg", "error reading the bucket index status", "err", err)
+		idxs = bucketindex.Status{Version: bucketindex.SyncStatusFileVersion, NonQueryableReason: bucketindex.Unknown}
+	}
+
+	idxs.Status = bucketindex.Ok
+	idxs.SyncTime = time.Now().Unix()
+
 	// Read the bucket index.
 	idx, err := bucketindex.ReadIndex(ctx, c.bucketClient, userID, c.cfgProvider, c.logger)
+
+	defer func() {
+		bucketindex.WriteSyncStatus(ctx, c.bucketClient, userID, idxs, userLogger)
```

Comment:
I am just curious. Why is error returned by `writeSyncStatus` not important to be handed her?

Reply:
I dont think we wanna stop anything if we were not able to write this file as it is only needed when other errors happens. But we can add some metrics if we think its needed.